AdvLoggerPkg/Library/AdvancedLoggerLib/Dxe/AdvancedLoggerLib.c

```diff
@@ -66,6 +66,17 @@ AdvancedLoggerWrite (
                     NULL,
                     (VOID **)&mLoggerProtocol
                     );
+
+    if (!EFI_ERROR (Status)) {
+      if ((mLoggerProtocol->Signature != ADVANCED_LOGGER_PROTOCOL_SIGNATURE) ||
+          (mLoggerProtocol->Version != ADVANCED_LOGGER_PROTOCOL_VERSION))
+      {
+        ASSERT (mLoggerProtocol->Signature == ADVANCED_LOGGER_PROTOCOL_SIGNATURE);

```

Comment:
Will this Assert end up calling this function and at this point we have never initialized mDebugPortProtocol.

Reply:
Without this change, a similar scenario exists, but it would _never_ initialize mDebugPortProtocol.